colossalai/zero/shard_utils/flat_shard_strategy.py

```diff
@@ -0,0 +1,52 @@
+from tokenize import group
+import torch
+from colossalai.zero.shard_utils import BaseShardStrategy
+import torch.distributed as dist
+from typing import List, Optional
+from torch import Tensor
+from colossalai.zero.sharded_param.sharded_tensor import ShardedTensor
+from torch._utils import _flatten_dense_tensors, _unflatten_dense_tensors
+from colossalai.zero.sharded_model._zero3_utils import get_shard
+from colossalai.utils import get_current_device
+
+
+class FlatShardStrategy(BaseShardStrategy):
+
+    """
+    FlatShardStategy flattens all tensors and shards the 1D tensor.
+    """
+
+    def __init__(self, process_group: Optional[dist.ProcessGroup] = None) -> None:
+        super().__init__(process_group)
+
+    def shard(self, tensor_list: List[Tensor]) -> List[Tensor]:
+        flat_tensor = _flatten_dense_tensors(tensor_list)
+        new_tensor_list = _unflatten_dense_tensors(flat_tensor, tensor_list)
+        tensor_shape_list = [tensor.shape for tensor in tensor_list]
+
+        # need to update the parameter data
+        for old, new in zip(tensor_list, new_tensor_list):
+            old.data = new.data
+
+        shard, _ = get_shard(flat_tensor, rank=self.local_rank, world_size=self.world_size)
+        shard.orig_shape_list = tensor_shape_list
+        return [shard]
+
+    def gather(self, tensor_list: List[ShardedTensor]) -> List[ShardedTensor]:
```

Comment:
The parameter type is diff from the abstract class.
def gather(self, tensor_list: List[Tensor]) -> List[Tensor]:

Reply:
Noted and will change soon.